src/rwe/grid/Grid.h

```diff
@@ -12,10 +12,10 @@ namespace rwe
 {
     struct GridCoordinates
     {
-        std::size_t x;
-        std::size_t y;
+        int x;
+        int y;
         GridCoordinates() = default;
-        GridCoordinates(size_t x, size_t y) : x(x), y(y)
+        GridCoordinates(int x, int y) : x(x), y(y)
         {
         }
```

Comment:
I wonder what to do here since `GridCoordinates` can now hold negative values (and can be default-constructed with them) which is invalid for this struct.

Reply:
I'm thinking we can put an assert in the constructor, same reasoning as for DiscreteRect.h